sys/net/crosslayer/ng_netif/ng_netif.c

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2015 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ */
+
+#include <stdio.h>
+
+#include <errno.h>
+#include "kernel_types.h"
+#include "net/ng_netif.h"
+
+static ng_netif_handler_t if_handler[] = {
```

Comment:
No. There are options that *must* be set on an interface, when it is used with IPv6 etc. Additionaly, there are actions that are done upon an interface if it is initialized, e.g. sending out neighbor solicitations, auto-address config packets, or DHCP packtes. These handlers bootstrap this behavior nicely if the interface is added post-IPv6/IPv4/… initialization. If they are initialized before-hand, the network protocol can just do it on it's own initialization.